app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -47,6 +54,31 @@ class SearchRobot {
     fun verifySearchSettings() = assertSearchSettings()
     fun verifySearchBarEmpty() = assertSearchBarEmpty()
 
+    fun verifyKeyboardVisibility() = assertKeyboardVisibility(isExpectedToBeVisible = true)
+    fun verifySearchEngineList() = assertSearchEngineList()
+    fun verifySearchEngineIcon(expectedText: String) = assertSearchEngineIconsList(expectedText)
+    fun verifySearchEngineIconsList() {
```

Comment:
I would remove this check in favor or a modified verifySearchIcon. What we want to be sure is that when the search engine is changed, that the new search is going to use that one. So, when we want to check that would be at this point:
<img width="893" alt="Captura de pantalla 2020-08-17 a las 13 26 44" src="https://user-images.githubusercontent.com/1897507/90392007-61434d80-e08e-11ea-9bf1-7cda798a0a84.png">

I would change the `verifySearchEngineIcon` like:

```
fun verifySearchEngineIcon(expectedText: String) {
        onView(withContentDescription(expectedText))
    }
```
And I would add a call to that function after changing the search engine each time...as commented in the smoketest file